File: front_end/src/features/users/components/user.model.ts
import * as Yup from "yup";
export const UserValidationSchema = Yup.object({
  name: Yup.string().max(30).required("Name is required"),
  email: Yup.string().email().required("Email is required"),
  role: Yup.string().required("Role is required"),
  phone: Yup.number().required("Phone is required"),
  address: Yup.string().required("Address is required"),
  status: Yup.string(),
});

File: Test_cases/backend-test-cases/src/models/user_model.py
from dataclasses import dataclass, field
from typing import Optional, Any, Union

@dataclass
class UserModel:
    name: str
    email: str
    role: Optional[Any] = None
    address: Optional[str] = None
    phone: Optional[Union[int, str]] = None
    status: Optional[str] = None
    _id: Optional[str] = None

    @classmethod
    def from_dict(cls, data: dict):
        # Filter out keys that are not in the dataclass
        valid_keys = cls.__dataclass_fields__.keys()
        filtered_data = {k: v for k, v in data.items() if k in valid_keys}
        return cls(**filtered_data)
